dlt/destinations/impl/lancedb/lancedb_client.py

```diff
@@ -190,6 +191,29 @@ def upload_batch(
             tbl.merge_insert(
                 id_field_name
             ).when_matched_update_all().when_not_matched_insert_all().execute(records)
+
+            # Remove orphaned parent IDs.
+            if parent_table_name:
+                try:
+                    parent_tbl = db_client.open_table(parent_table_name)
```

Comment:
this will not work, there is no guarantee that the parent table is fully loaded (with all possible load jobs) at the point in time of the execution of the child table job. If we do it this way, we will need table chain followup jobs, similar to the way merge jobs are done. You can have a look at the default sql client for that.

Reply:
Fixed in `LanceDBRemoveOrphansJob`